### PR TITLE
ALSA: hda: intel-dsp-config: add PCI id for TGL-H

### PR DESCRIPTION
**Pull request by bardliao**
SOF driver will be selected if DMIC or SoundWire is used.

Signed-off-by: Bard Liao <bard.liao@intel.com>

Fixes: #2661

### CODE REVIEW

File: sound/hda/intel-dsp-config.c [hunk 1]
```diff
@@ -307,6 +307,10 @@ static const struct config_entry config_table[] = {
 		.flags = FLAG_SOF | FLAG_SOF_ONLY_IF_DMIC_OR_SOUNDWIRE,
 		.device = 0xa0c8,
 	},
+	{
+		.flags = FLAG_SOF,
```
Comment (xiulipan) [inline sound/hda/intel-dsp-config.c:298]:
```suggestion
		.flags = FLAG_SOF | FLAG_SOF_ONLY_IF_DMIC_OR_SOUNDWIRE,
```

**Review by xiulipan — CHANGES_REQUESTED** [review]

File: sound/hda/intel-dsp-config.c (hunk 1 above)
Comment (bardliao) [reply sound/hda/intel-dsp-config.c:298]: @xiulipan It was my first version. But after talking with @mengdonglin , it seems that SST driver doesn't support TGL-H. In 
 other words, SOF driver is the only audio driver which supports TGL-H. That's why I changed the flag to FLAG_SOF.

Comment (xiulipan) [reply sound/hda/intel-dsp-config.c:298]: @bardliao @mengdonglin do we got SST support on TGL-U?

Comment (mengdonglin) [reply sound/hda/intel-dsp-config.c:298]: @xiulipan We don't have SST support on TGL-U or TGL-H

Comment (xiulipan) [reply sound/hda/intel-dsp-config.c:298]: @bardliao @mengdonglin Then I am thinking we may also need a clean up for above few lines for TGL-U (or EHL)
We should only have `.flags = FLAG_SOF,` for them too, right?

Comment (bardliao) [reply sound/hda/intel-dsp-config.c:298]: > @bardliao @mengdonglin Then I am thinking we may also need a clean up for above few lines for TGL-U (or EHL)
> We should only have `.flags = FLAG_SOF,` for them too, right?

Yes, I will do it.

**Review by xiulipan — APPROVED** [review]
LGTM

**Review by xiulipan — CHANGES_REQUESTED** [review]
@bardliao maybe also a quick clean up for EHL?

**Review by RanderWang — APPROVED** [review]

**Comment by bardliao** [comment]
> @bardliao maybe also a quick clean up for EHL?

@xiulipan Done

**Review by libinyang — APPROVED** [review]

**Review by xiulipan — APPROVED** [review]
Thanks for the fix.

**Comment by kv2019i** [comment]
@bardliao @xiulipan @libinyang I'm afraid this is not quite correct. Even for TGL, EHL and newer, the legacy driver should still be the default, unless Soundwire or DMIC is detected.

**Comment by xiulipan** [comment]
@mengdonglin @plbossart Any clarify here? On TGL+ platforms, will SOF be default or SOF only if DMIC and SDW.

**Comment by bardliao** [comment]
> @bardliao @xiulipan @libinyang I'm afraid this is not quite correct. Even for TGL, EHL and newer, the legacy driver should still be the default, unless Soundwire or DMIC is detected.

@kv2019i Oops, I forgot legacy HDA driver. Can we revert 7c86362591125edfc40825be6aa63146662667f9?

**Comment by bardliao** [comment]
The fixup PR is https://github.com/thesofproject/linux/pull/2672. I am thinking someday we may need to detect I2S as well since SOF will be the default if I2S is using.

File: sound/hda/intel-dsp-config.c [hunk 2]
```diff
@@ -293,26 +293,13 @@ static const struct config_entry config_table[] = {
...
-				.matches = {
-					DMI_MATCH(DMI_SYS_VENDOR, "Google"),
-				}
-			},
-			{}
-		}
-	},
-	{
-		.flags = FLAG_SOF | FLAG_SOF_ONLY_IF_DMIC_OR_SOUNDWIRE,
-		.device = 0xa0c8,
```
Comment (plbossart) [inline sound/hda/intel-dsp-config.c:308]: No that's not correct. If we don't have DMIC/SoundWire, and it's not a Chromebook, then we want to use the legacy HDaudio driver.
The Chromebook does not rely on NHLT so the DMIC detection will fail, so it needs a dedicated quirk.

This needs to be reverted IMHO.

File: sound/hda/intel-dsp-config.c [hunk 3]
```diff
@@ -294,6 +294,10 @@ static const struct config_entry config_table[] = {
 		.flags = FLAG_SOF,
 		.device = 0xa0c8,
 	},
+	{
+		.flags = FLAG_SOF,
+		.device = 0x43c8,
+	},
```
Comment (plbossart) [inline sound/hda/intel-dsp-config.c:300]: same comment as above, this will unconditionally select SOF. This is not what we want.

**Review by plbossart — COMMENTED** [review]
Revert needed!

**Pull request merged**